extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -76,6 +76,10 @@ export class ApplicationInsights extends BaseTelemetryPlugin implements IAppInsi
         config.enableAutoRouteTracking = stringToBoolOrDefault(config.enableAutoRouteTracking);
         config.namePrefix = config.namePrefix || "";
 
+        config.enableDebug = stringToBoolOrDefault(config.enableDebug);
+        config.disableFlushOnBeforeUnload = stringToBoolOrDefault(config.disableFlushOnBeforeUnload);
+        config.disableFlushOnUnload = stringToBoolOrDefault(config.disableFlushOnUnload, config.disableFlushOnBeforeUnload);
```

Comment:
@MSNev Sorry Nev for the confusion... I see what's going on here. 

For other configurations I added before, they're not analytics extension related and should not bring in here. 

`maxBatchSizeInBytes, maxBatchInterval, isBeaconApiDisabled, enableSessionStorageBuffer, isRetryDisabled, disableTelemetry` are `sender config` and initialized [here](https://github.com/microsoft/ApplicationInsights-JS/blob/3d4a6fa42a7a25ceedbf3131598819249797ad47/channels/applicationinsights-channel-js/src/Sender.ts#L81). And when user passes in user defined config, it receives [here](https://github.com/microsoft/ApplicationInsights-JS/blob/3d4a6fa42a7a25ceedbf3131598819249797ad47/channels/applicationinsights-channel-js/src/Sender.ts#L220).

 `maxAjaxCallsPerView , disableCorrelationHeaders, enableCorsCorrelation, correlationHeaderExcludedDomains, disableAjaxTracking` are `remote dependency extension config` , initialized [here ](https://github.com/microsoft/ApplicationInsights-JS/blob/3d4a6fa42a7a25ceedbf3131598819249797ad47/extensions/applicationinsights-dependencies-js/src/ajax.ts#L153)and received user defined [here](https://github.com/microsoft/ApplicationInsights-JS/blob/3d4a6fa42a7a25ceedbf3131598819249797ad47/extensions/applicationinsights-dependencies-js/src/ajax.ts#L218). 

And `disableCookiesUsage` is handled by the cookie manager you created. 

But for the above three configs, this is the place to initialize them. And on line 505, getDefaultConfig is missing the param `config` so the user defined configs dropped!

I think we need this config.[key] pattern because we don't need all keys from config, like "extensionConfig" is in config object too but we only need to initialize some of the configs... does that make sense?